src/pages/dashboard/globalWebsite/contact/index.js

```diff
@@ -1,7 +1,199 @@
+
+  const parsed = qs.parse(location.search);
+
+  const dataLimit = 10;
+
+  const fetchAllContacts = async () => {
+    try {
+      const { data } = await ContactService.getAllContacts({
+        keyword: parsed.keyword,
+        paged: parsed.page,
+        start_date: parsed.start_date,
+        end_date: parsed.end_date,
+        limit: dataLimit,
+      });
+      setTotalItems(data.count);
+      setListContact(data.data);
+    } catch (error) {
+      errHandler(error);
+    }
+  };
+
+  const [getAllContacts, loading] = useLoading(fetchAllContacts);
+
+  function handleOnChangeSearch({ target: { value } = {} } = {}) {
+    setKeyword(value);
+  }
+  useDebounce(() => filterContacts('keyword', keyword), 500, [keyword]);
+
+  function handleOnChangeDate(value) {
+    const [start, end] = value || [];
+    const startDate = start && moment(start).format(DATE_FORMAT);
+    const endDate = start && moment(end).format(DATE_FORMAT);
+    filterContacts('start_date', startDate);
+    filterContacts('end_date', endDate);
+  }
+
+  function filterContacts(param, val) {
+    if (val === '' || val === 0) {
+      delete parsed[param];
+    } else {
+      parsed[param] = val;
+    }
+
+    const search = qs.stringify(parsed);
+    history.push({
+      pathname: '/global-website/contact',
+      search,
+    });
+  }
+
+  function resetFilter() {
+    const search = qs.stringify(
+      omit(parsed, ['keyword', 'page', 'type', 'start_date', 'end_date'])
+    );
+    history.push({
+      pathname: '/points',
+      search,
+    });
+  }
+
+  async function deleteContact(id) {
+    try {
+      await ContactService.deleteContact(id);
+      getAllContacts();
+    } catch (error) {
+      errHandler(error);
+    }
+  }
+
+  useEffect(() => {
+    if (parsed.page) {
+      if (parsed.page < 1) {
+        parsed.page = 1;
+        const stringified = qs.stringify(parsed);
+        history.push({
+          pathname: '/global-website/contact',
+          search: stringified,
+        });
+      } else setPage(parsed.page);
+    } else setPage(1);
+
+    if (parsed.keyword) setKeyword(parsed.keyword);
+    else setKeyword('');
+
+    if (parsed.start_date) {
+      setDateRange((prev) => [moment(parsed.start_date, DATE_FORMAT), prev[1]]);
+    } else setDateRange((prev) => [undefined, prev[1]]);
+
+    if (parsed.end_date) {
+      setDateRange((prev) => [prev[0], moment(parsed.end_date, DATE_FORMAT)]);
+    } else setDateRange((prev) => [prev[0], undefined]);
+
+    getAllContacts();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [parsed.end_date, parsed.keyword, parsed.page, parsed.start_date]);
+
+  const columns = [
+    {
+      key: 'id',
+      title: 'ID',
+      dataIndex: 'id',
+    },
+    {
+      key: 'email',
+      title: 'Email',
+      dataIndex: 'email',
+    },
+    {
+      key: 'createdAt',
+      title: 'Date Transaction',
+      render: ({ createdAt } = {}) => moment(createdAt).format(DATE_FORMAT),
```

Comment:
createdAt nya yg dikasi default value ya..
soalnya paramnya udah di destruct, jadi kalau param kosong, harusnya sih tetep return { createdAt: undefined }

tapi kalau createdAt nya yg ada def value, pas paramnya kosong returnnya { createdAt: `defaultValue` }
```suggestion
      render: ({ createdAt = '' }) => moment(createdAt).format(DATE_FORMAT),
```

Reply:
Baik Mba

Reply:
Saya mikirnya return object kosong sih Mba sama seperti kode kode sebelumnya, tapi empty string juga harusnya bisa sih

Reply:
hmm as mentioned before, kalau udah di deconstruct, harusnya ga bisa return object kosong walaupun value yg di pass ga ada createdAt.

Reply:
`kode kode sebelumnya` yg kamu maksud itu mungkin karna object yg di passed itu ada objectnya lagi cmiiw
misal:

```
const data = {
text: 'asdasd',
obj: {
a: 1,
b: 2,
}
}

function misal({
text = 'default value a',
obj = {} ---> empty object buat default value b karna b adalah object
}) {...}
```

Reply:
jadi default valuenya bukan di deconstructed obj nya, tapi di dalam nya